scipy/linalg/_decomp_svd.py

```diff
@@ -299,6 +303,101 @@ def diagsvd(s, M, N):
         raise ValueError("Length of s must be M or N.")
 
 
+def higher_order_svd(
+    a: npt.ArrayLike, *, compact_rtol: float | None = None, check_finite: bool = True
+) -> tuple[list[np.ndarray], np.ndarray]:
+    """Higher-order SVD (HOSVD)
+
+    Factorizes the M-D tensor `a` into a list of M matrices ``U_k`` containing
+    the left singular vectors of the unfolded tensor with respect to the
+    ``k``\\ th axis and a core tensor ``S`` such that
+    :math:`a = (U_1, U_2, \\dots, U_M) \\cdot S`.
+
+    Parameters
+    ----------
+    a : array_like
+        Tensor to decompose
+    compact_rtol : float, optional
```

Comment:
I'm happy to stick with `compact_rtol`

Reply:
Fine for me, too! :)

Reply:
As a compromise with https://github.com/scipy/scipy/pull/14284#issuecomment-2525051965, what do you think of renaming this to just `rtol`? Note that `pinv` and `pinvh` use `atol` and `rtol` for the threshold for significance of singular values, so maybe it is better to just roll with it rather than trying to be more specific. That said, `orth` and `null_space`, which are also closely related, use `rcond`. Perhaps we just call this `rtol` now, and after this is merged, open an issue about standardizing the name? We can use a decorator that will make it easy to change the name of `rcond` in a backward compatible way, if that's what we decide.

https://github.com/scipy/scipy/pull/14284#issuecomment-2525051965 also gives the answer to [my question](https://github.com/scipy/scipy/pull/14284#discussion_r1735040827) from August. So one possibility is that we leave `rtol` required for now, then open an enhancement issue about adding a `compact` or similar argument: when `compact=True`, `rtol` would no longer be required, and the default could be computed according to the suggestion. We could add `atol` at that time, too. (The other option is doing both of these things now.)

With that in mind, new thoughts?